### PR TITLE
WIP: Use Rust Setup Action

### PR DESCRIPTION
**Pull request by H00N24**
## 🚀 Pull Request Overview

Please provide a brief summary of your changes and the motivation behind them.

- **Related Issues/Tickets**: 
- **Type of Change** (bugfix, feature, refactor, etc.): 
- **Breaking Changes?**: 

---

### 📜 Tests Checklist

Before submitting this PR, ensure that all tests pass **and** meet the following conditions:

| Category                  | Status                          | Compilation Comment | Results Review |
|---------------------------|----------------------------------|----------------------|----------------|
| **fail_to_start**         | ❌ Failed with compilation error | 🟢 Yes               | 🔴 No          |
| **violated_rules**        | ✅ Passed, with violations found | 🔴 No                | 🟢 Yes         |
| **verified_rules**        | ✅ Passed without violations     | 🔴 No                | 🟢 Yes         |
| **solana_violated_rules** | ✅ Passed, with violations found | 🔴 No                | 🟢 Yes         |

> Please verify that your changes meet the above conditions for each category of tests.  
> If something doesn't match, investigate before submitting the PR.

---

### 📋 Additional Notes (Optional)

- Any risks, edge cases, or implementation notes for reviewers:

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -238,13 +265,13 @@ runs:
...
     - name: Install rust for solana
-      if: ${{ inputs.ecosystem == 'solana' }}
-      shell: bash
-      run: |
-        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.86.0
-        . $HOME/.cargo/env
-        cargo install cargo-certora-sbf
-        cargo certora-sbf --no-build
+      if: ${{ inputs.ecosystem == 'solana' && inputs.rust-setup == 'true' }}
+      uses: Certora/rust-setup-action@v1
```
[the remark this answers is not in the document]
Comment (H00N24) [reply action.yml]: Yes.

**Review by ozcodes — APPROVED** [review]
